code/drasil-docLang/lib/Drasil/DocumentLanguage.hs

```diff
@@ -74,78 +74,47 @@ mkDoc dd comb si@SI {_sys = sys, _kind = kind, _authors = docauthors} =
--}
-
-{- FIXME: See #2775
--- Fill in term map from all concepts and quantities.
-fillTermMap :: SystemInformation -> SystemInformation
-fillTermMap si = si2
-  where
-    -- Get current contents of si
-    chkdb = si ^. sysinfodb
-    -- extract everything that could possibly lead to an 'IdeaDict'
-    symbs    = map (fst.snd) $ Map.assocs $ symbolTable chkdb
-    trms     = map (fst.snd) $ Map.assocs $ termTable chkdb
-    concepts = map (fst.snd) $ Map.assocs $ defTable chkdb
-
-    -- TODO: Uncomment these when the second part of #2775 is resolved.
-    -- Some Definitions and models overwrite the term for a given UID.
-    -- We don't really want this behaviour, so it should be resolved by
-    -- changing some of the constructors for ModelKind found in drasil-theory
-
-    ddefs   = map (fst.snd) $ Map.assocs $ chkdb ^. dataDefnTable
-    gdefs   = map (fst.snd) $ Map.assocs $ chkdb ^. gendefTable
-    imods   = map (fst.snd) $ Map.assocs $ chkdb ^. insmodelTable
-    tmods   = map (fst.snd) $ Map.assocs $ chkdb ^. theoryModelTable
-    concIns = map (fst.snd) $ Map.assocs $ chkdb ^. conceptinsTable
-    -- fill in the appropriate chunkdb field
-    chkdb2 = chkdb {termTable = termMap $ nub $ map nw symbs ++ map nw trms
-      ++ map nw concepts ++ map nw concIns
-      ++ map nw ddefs ++ map nw gdefs ++ map nw imods ++ map nw tmods}
-    -- return the filled in system information
-    si2 = set sysinfodb chkdb2 si
-    -}
-
 -- | Fill in the 'Section's and 'LabelledContent' maps of the 'ChunkDB' from the 'SRSDecl'.
 fillSecAndLC :: SRSDecl -> SystemInformation -> SystemInformation
 fillSecAndLC dd si = si2
   where
     -- Get current contents of si
     chkdb = si ^. sysinfodb
     -- extract sections and labelledcontent
-    allSections = concatMap findAllSec $ mkSections si $ mkDocDesc si dd
+    allSections = concatMap findAllSec $ mkSections si $ mkDocDesc si dd -- FIXME: `mkSections` on something particularly large that is immediately discarded is a sign that we're doing something wrong. That's in addition to `mkDocDesc`...
     allLC = concatMap findAllLC allSections
     existingLC = map (fst.snd) $ Map.assocs $ chkdb ^. labelledcontentTable
     -- fill in the appropriate chunkdb fields
     chkdb2 = set labelledcontentTable (idMap $ nub $ existingLC ++ allLC) chkdb
     -- return the filled in system information
     si2 = set sysinfodb chkdb2 si
+
     -- Helper and finder functions
     findAllSec :: Section -> [Section]
     findAllSec s@(Section _ cs _) = s : concatMap findAllSubSec cs
+
     findAllSubSec :: SecCons -> [Section]
     findAllSubSec (Sub s) = findAllSec s
     findAllSubSec _ = []
+
     findAllLC :: Section -> [LabelledContent]
     findAllLC (Section _ cs _) = concatMap findLCSecCons cs
+
     findLCSecCons :: SecCons -> [LabelledContent]
     findLCSecCons (Sub s) = findAllLC s
-    findLCSecCons (Con (LlC lblcons)) = [lblcons]
+    findLCSecCons (Con (LlC lblcons@(LblC {_ctype = Figure {}}))) = [lblcons]
```

Comment:
TLDR; Of the `LabelledContent` we extract from the SRS, we only use the `Figure`-type `LabelledContent`. The manually-included `LabelledContent` is also used.